### PR TITLE
PWGHF: Calculate variables using updated tracks after KF vertex fit.

### PR DESCRIPTION
**Pull request by Yunfan-Liu**
- Now calculate variables using updated tracks after KF vertex fit instead of initial tracks.

- Add cuts to make sure Chi2 and Ndf are greater than 0.

### CODE REVIEW

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx [hunk 1]
```diff
@@ -680,109 +680,146 @@ struct HfCandidateCreatorXic0Omegac0 {
...
+        LOG(debug) << "Failed to construct Omega from V0 and bachelor track: " << e.what();
         continue;
       }
       float massCasc, sigCasc;
       KFOmega.GetMass(massCasc, sigCasc);
+      // err_massOmega > 0
+      if (sigCasc <= 0)
+        continue;
+      if (TMath::Abs(massCasc - MassOmegaMinus) > massToleranceCascade)
+        continue;
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx:714]:
```suggestion
      if (std::abs(massCasc - MassOmegaMinus) > massToleranceCascade)
        continue;
```

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx [hunk 2]
```diff
@@ -680,109 +680,146 @@ struct HfCandidateCreatorXic0Omegac0 {
...
+      // chi2>0 && NDF>0
+      if (KFOmega.GetNDF() <= 0 || KFOmega.GetChi2() <= 0)
+        continue;
+      kfOmegac0Candidate.chi2GeoCasc = KFOmega.GetChi2();
+      KFParticle KFOmega_m = KFOmega;
+      KFOmega_m.SetNonlinearMassConstraint(o2::constants::physics::MassOmegaMinus);
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx:720]: What does the suffix `_m` stand for? Please choose a more meaningful name and avoid underscores.

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx [hunk 3]
```diff
@@ -680,109 +680,146 @@ struct HfCandidateCreatorXic0Omegac0 {
...
-      registry.fill(HIST("hKFParticleV0Chi2"), KFV0.GetChi2());
-      if (kfUseV0MassConstraint) {
-        KFV0.SetNonlinearMassConstraint(o2::constants::physics::MassLambda);
-      }
-
+      // err_mass>0 of Lambda
+      if (sigLam <= 0)
+        continue;
+      // chi2>0 && NDF>0 for selecting Lambda
+      if ((KFV0.GetNDF() <= 0 || KFV0.GetChi2() <= 0))
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx:687]:
```suggestion
      if ((kfV0.GetNDF() <= 0 || kfV0.GetChi2() <= 0))
```
Please use lowerCamelCase for names of variables.

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx [hunk 4]
```diff
@@ -680,109 +680,146 @@ struct HfCandidateCreatorXic0Omegac0 {
...
+      if (kfUseV0MassConstraint) {
+        KFParticle KFV0 = KFV0_m;
+      }
 
       //__________________________________________
-      //*>~<* step 2 : reconstruc cascade(Omega) with KF
-      KFParticle kfpV0 = KFV0;
-      const KFParticle* OmegaDaugthers[2] = {&kfpBachKaon, &kfpV0};
+      //*>~<* step 2 : reconstruct cascade(Omega) with KF
+      const KFParticle* OmegaDaugthers[2] = {&kfpBachKaon, &KFV0};
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx:698]:
```suggestion
      const KFParticle* omegaDaugthers[2] = {&kfPBachKaon, &kfV0};
```

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx (hunk 2 above)
Comment (Yunfan-Liu) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: It's done. `_m` stands for MassConstrained KFparticle, now change variables name as : KFV0_m -> kfV0MassConstrained, KFOmega_m -> kfOmegaMassConstrained

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx (hunk 3 above)
Comment (Yunfan-Liu) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: It's done.

File: PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx (hunk 4 above)
Comment (Yunfan-Liu) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: It's done.

Comment (vkucera) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: Doesn't seem so.

Comment (Yunfan-Liu) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: Sorry, I missed the `OmegaDaugthers`. It's done now.

Comment (vkucera) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: Thanks. Why does `kfpBachKaon` start with `kfp` and `kfV0` with `kf`? What is the meaning of `p`?

Comment (vkucera) [reply PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx]: OK, thanks. That looks much better.

**Review by vkucera — APPROVED** [review]

**Pull request merged**